crates/assists/src/utils.rs

```diff
@@ -356,6 +355,22 @@ pub(crate) fn find_impl_block(impl_def: ast::Impl, buf: &mut String) -> Option<T
     Some(start)
 }
 
+/// Find the end of the `impl` block for the given `ast::Impl`.
+//
+// FIXME: this partially overlaps with `find_struct_impl`
+pub(crate) fn find_impl_block_end(impl_def: ast::Impl, buf: &mut String) -> Option<TextSize> {
+    buf.push('\n');
+    let end = impl_def
+        .syntax()
+        .descendants_with_tokens()
+        .filter(|t| t.kind() == T!['}'])
+        .last()?
```

Comment:
```suggestion
        .assoc_item_list()
        .and_then(|it| it.r_curly_token())?
```
I think this should do the same but without traversing all the descendants. Same can be applied to `find_impl_block_start` with `l_curly_token`.

Reply:
Oh neat; done!